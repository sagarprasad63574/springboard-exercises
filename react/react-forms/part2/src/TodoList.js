import React, {useState} from 'react';
import Todo from './Todo.js';
import NewTodoForm from './NewTodoForm.js';

const TodoList = () => {
    const [todos, setTodos] = useState([]);
    const addTodo = (newTodo) => {
        setTodos(todos => [...todos, newTodo]);
    }

    const removeTodo = (id) => {
        setTodos(todos => todos.filter(todo => todo.id !== id));
    }

    return (
        <div>
            <h3>Todo App</h3>
            <NewTodoForm addTodo={addTodo} />
            <div>
                {todos.map(
                    ({id, text}) => 
                    <Todo 
                        key={id}
                        id={id}
                        text={text} 
                        handleRemove={removeTodo} />)}
            </div>
        </div>
    )
}

export default TodoList; 