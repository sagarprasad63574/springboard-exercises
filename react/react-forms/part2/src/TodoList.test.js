import React from "react";
import { render, fireEvent, wait } from "@testing-library/react";
import TodoList from "./TodoList";

function addTodo(todolist, text = "new todo") {
  const taskInput = todolist.getByLabelText("Text");
  fireEvent.change(taskInput, { target: { value: text } });
  const button = todolist.getByText("Add Todo");
  fireEvent.click(button);
}

it("renders without crashing", function() {
  render(<TodoList />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("can add a new todo", function() {
  const todolist = render(<TodoList />);

  expect(todolist.queryByText("X")).not.toBeInTheDocument();

  addTodo(todolist);

  const removeButton = todolist.getByText("X");
  expect(removeButton).toBeInTheDocument();
  expect(todolist.queryByText("new todo")).toBeInTheDocument();

});

it("can remove a todo", function() {
  const todolist = render(<TodoList />);
  addTodo(todolist);

  const removeButton = todolist.getByText("X");

  fireEvent.click(removeButton);
  expect(removeButton).not.toBeInTheDocument();
});
