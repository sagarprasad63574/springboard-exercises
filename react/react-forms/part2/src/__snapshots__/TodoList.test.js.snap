// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Todo App
    </h3>
    <div>
      <form>
        <label
          for="text"
        >
          Text
        </label>
        <input
          id="text"
          name="text"
          placeholder="text"
          type="text"
          value=""
        />
        <button>
          Add Todo
        </button>
      </form>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
